src/libcrun/cgroup-setup.c

```diff
@@ -104,6 +104,15 @@ initialize_cpuset_subsystem_rec (char *path, size_t path_len, char *cpus, char *
         }
     }
 
+  /* If we know the resources, use them, instead of initializing with the full set, only to revert it later. */
+  if (resources && resources->cpu)
+    {
+      if (resources->cpu->cpus)
+        cpus = xstrdup (resources->cpu->cpus);
+      if (resources->cpu->mems)
+        mems = xstrdup (resources->cpu->mems);
+    }
```

Comment:
shouldn't this be done only for the leaf node (or possibly its parent as well)?

What happens if we use a cgroup like `a/b/c/d/e` and only `a/` exists?  Do we override the settings for `a/b`,  `a/b/c`, and `a/b/c/d`?

Reply:
it is probably enough to add a check and do not override the file if there is already a value specified

Reply:
AFAICT it only overrides the empty cgroups. For instance, with the slice hierarchy of a/b/c where a is the pre-initialized pod slice, only b and c are populated as expected